apps/desktop/src-tauri/src/qdrant.rs

```diff
@@ -0,0 +1,256 @@
+    # Minimal size (in KiloBytes) of vectors for additional payload-based indexing.
+    # If payload chunk is smaller than `full_scan_threshold_kb` additional indexing won't be used -
+    # in this case full-scan search should be preferred by query planner and additional indexing is not required.
+    # Note: 1Kb = 1 vector of size 256
+    full_scan_threshold_kb: 10000
+    # Number of parallel threads used for background index building. If 0 - auto selection.
+    max_indexing_threads: 0
+
+service:
+
+  # Maximum size of POST data in a single request in megabytes
+  max_request_size_mb: 32
+
+  # Number of parallel workers used for serving the api. If 0 - equal to the number of available cores.
+  # If missing - Same as storage.max_search_threads
+  max_workers: 0
+
+  # Host to bind the service on
+  host: 127.0.0.1
+
+  # HTTP port to bind the service on
+  http_port: 6333
+
+  # gRPC port to bind the service on.
+  # If `null` - gRPC is disabled. Default: null
+  # Uncomment to enable gRPC:
+  grpc_port: 6334
+
+  # Enable CORS headers in REST API.
+  # If enabled, browsers would be allowed to query REST endpoints regardless of query origin.
+  # More info: https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS
+  # Default: true
+  enable_cors: true
+
+cluster:
+  # Use `enabled: true` to run Qdrant in distributed deployment mode
+  enabled: false
+
+  # Configuration of the inter-cluster communication
+  p2p:
+    # Port for internal communication between peers
+    port: 6335
+
+  # Configuration related to distributed consensus algorithm
+  consensus:
+    # How frequently peers should ping each other.
+    # Setting this parameter to lower value will allow consensus
+    # to detect disconnected nodes earlier, but too frequent
+    # tick period may create significant network and CPU overhead.
+    # We encourage you NOT to change this parameter unless you know what you are doing.
+    tick_period_ms: 100
+"#;
+
+#[derive(Default)]
+pub(super) struct QdrantSupervisor {
+    child: Option<Child>,
+}
+
+impl<R> Plugin<R> for QdrantSupervisor
+where
+    R: Runtime,
+{
+    fn name(&self) -> &'static str {
+        "qdrant"
+    }
+
+    fn initialize(
+        &mut self,
+        app: &tauri::AppHandle<R>,
+        _config: serde_json::Value,
+    ) -> tauri::plugin::Result<()> {
+        let cache_dir = app.path_resolver().app_cache_dir().unwrap();
+        let qdrant_dir = cache_dir.join("qdrant");
+        let qd_config_dir = qdrant_dir.join("config");
+        create_dir_all(&qd_config_dir).unwrap();
+        write(
+            qd_config_dir.join("config.yaml"),
+            QDRANT_CONFIG
+                .replace(
+                    "{{ STORAGE }}",
+                    &qdrant_dir.join("storage").to_string_lossy(),
+                )
+                .replace(
+                    "{{ SNAPSHOTS }}",
+                    &qdrant_dir.join("snapshots").to_string_lossy(),
+                ),
+        )
+        .unwrap();
```

Comment:
there seems to be some trivial templating going on here, we can avoid using string replacements here by doing two things:

- put the template config into `config.yaml` and use rust's format string syntax, i.e., use `{storage}` for template holes
- achieve templating with the `format!` and `include_str!` macros, `format!(inculude_str!("./config.yaml"), storage = storage, snapshots = snapshots)`

Reply:
https://github.com/BloopAI/bloop/pull/96